src/coreclr/vm/threads.cpp

```diff
@@ -893,7 +898,7 @@ void DestroyThread(Thread *th)
     }
     CONTRACTL_END;
 
-    _ASSERTE (th == GetThread());
+    _ASSERTE (th == GetThreadNULLOk());
```

Comment:
I generally kept the pattern `_ASSERTE (th == GetThreadNULLOk());` because I assumed the invariant guarde by the assert is stronger than just having a thread object. 
`th` in thse cases are typically not NULL and that is enforced at the place where `th` is stored.

I guess it would work either way.